scripts/js/lib/api/objectsInv.ts

```diff
@@ -41,8 +41,29 @@ const ENTRIES_TO_EXCLUDE = [
   /^legacy_release_notes(\.html)?(?=#|$)/,
 ];
 
-function shouldExcludePage(uri: string): boolean {
-  return ENTRIES_TO_EXCLUDE.some((condition) => uri.match(condition));
+const C_API_BASE_PATH = "cdoc/";
+
+type PackageLanguage = "Python" | "C" | "any";
+
+function shouldIncludeEntry(
+  entry: ObjectsInvEntry,
+  packageLanguage: PackageLanguage,
+): boolean {
+  if (ENTRIES_TO_EXCLUDE.some((condition) => entry.uri.match(condition))) {
+    return false;
+  }
+
+  // `group__` and `struct_` entries are from doxygen and are not present in the final pages.
+  if (entry.name.startsWith("group__")) return false;
+  if (entry.name.startsWith("struct_")) return false;
+
+  // Not specified: Include both C and Python entries.
+  // This happens during link checking.
+  if (packageLanguage === "any") return true;
```

Comment:
Imo the comment is necessary anymore because this new factoring is self-documenting

```suggestion
  if (packageLanguage === "any") return true;
```